### PR TITLE
Actually run `go build` in docker (cd&rollout)

### PR DESCRIPTION
**Pull request by sven-urbanski-freiheit-com**
- Removes the "backend" docker container (it was redundant).
- cd & rollout services now actually build go code in docker. Before, the binary was build on the host, and then only copied into the docker image.

### CODE REVIEW

**Review by leandro-freiheit — APPROVED** [review]

**Comment by hannesg** [comment]
I think the name `swissknife` is really bad. I sounds like "does everything for that we don't have a name". Why not keep builder?

File: Makefile.variables [hunk 1]
```diff
@@ -16,3 +16,9 @@
 
 VERSION?=$(shell git describe --always --long --tags || echo 0.0.1)
 KUBERPULT_BUILDER?=europe-west3-docker.pkg.dev/fdc-public-docker-registry/kuberpult/kuberpult-builder:1.5.0-7-g74d2058
+
+# PARENT_CONTAINER_VERSION is the docker tag of the parent image. e.g. for a go service, this would be the "go" image (could be called "swissknife", "builder", etc)
+# For local development, you can set it to latest
+#PARENT_CONTAINER_VERSION=latest
```
Comment (ahmed-nour-fdc) [inline Makefile.variables:22]: Can we remove this commented line?

Comment (sven-urbanski-freiheit-com) [reply Makefile.variables]: done

File: services/rollout-service/Makefile [hunk 2]
```diff
@@ -71,8 +71,8 @@ test: test-go
 test-go: $(GO_FILES) proto
 	$(GO) test ./...
 
-docker: bin/main
-	docker build -t $(IMAGENAME) .
+docker:
+	docker build -t $(IMAGENAME) ../.. -f Dockerfile --build-arg="PARENT_CONTAINER_VERSION=$(PARENT_CONTAINER_VERSION)"
```
[the remark this answers is not in the document]
Comment (sven-urbanski-freiheit-com) [reply services/rollout-service/Makefile]: fixed.
I supply this in the exec-planner GH action, so we have the exact docker image name

File: Makefile.variables [hunk 3]
```diff
@@ -16,3 +16,8 @@
 
 VERSION?=$(shell git describe --always --long --tags || echo 0.0.1)
 KUBERPULT_BUILDER?=europe-west3-docker.pkg.dev/fdc-public-docker-registry/kuberpult/kuberpult-builder:1.5.0-7-g74d2058
+
+# PARENT_CONTAINER is the full docker image name of the parent image. e.g. for a go service, this would be the "go" image (could be called "builder", etc)
+# For local development, you can set it to latest
```
Comment (ahmed-nour-fdc) [inline Makefile.variables:21]: Is this comment needed anymore? It references `PARENT_CONTAINER` which does not exist in this file.

File: services/cd-service/Dockerfile [hunk 4]
```diff
@@ -1,7 +1,30 @@
+ARG PARENT_CONTAINER
+FROM ${PARENT_CONTAINER} as builder
```
Comment (ahmed-nour-fdc) [inline services/cd-service/Dockerfile:2]: In `docker-compose.yaml` the `PARENT_CONTAINER` arg is set to `latest` which is an incorrect name for the base image.

File: docker-compose.yml [hunk 5]
```diff
@@ -1,12 +1,18 @@
...
-      context: infrastructure/docker/backend
+  cd-service:
+    build:
+      context: .
+      dockerfile: services/cd-service/Dockerfile
       args:
         - UID=$USER_UID
-    container_name: kuberpult-cd-service
+        - DIR=services/cd-service
+        - PARENT_CONTAINER=latest
```
Comment (ahmed-nour-fdc) [inline docker-compose.yml:9]: The dockerfiles expect `PARENT_CONTAINER` to be the full name of the image not `latest`

File: Makefile.variables (hunk 3 above)
Comment (sven-urbanski-freiheit-com) [reply Makefile.variables]: done

File: docker-compose.yml (hunk 5 above)
Comment (sven-urbanski-freiheit-com) [reply docker-compose.yml:9]: done

**Pull request merged**